packages/atomic/src/components/search/atomic-search-interface/atomic-aria-live.tsx

```diff
@@ -1,27 +1,34 @@
-import {Component, h, Host, State, Element, Method} from '@stencil/core';
 import {
-  FindAriaLiveEventArgs,
-  findAriaLiveEventName,
-} from '../../../utils/accessibility-utils';
+  Component,
+  h,
+  Host,
+  State,
+  Element,
+  Method,
+  Listen,
+} from '@stencil/core';
+import {FindAriaLiveEventArgs} from '../../../utils/accessibility-utils';
+import {buildDebouncedQueue} from '../../../utils/debounce-utils';
+import {randomID} from '../../../utils/utils';
+
+type Regions = {[regionName: string]: {assertive: boolean; message: string}};
 
 /**
  * The `atomic-aria-live` component notifies screen readers of changes in the search interface.
- *
- * We do not recommend dynamically adding/removing this component from the page. It is better to have this component during the initial render of the page.
  */
 @Component({
   tag: 'atomic-aria-live',
   shadow: false,
 })
 export class AtomicAriaLive {
   @Element() private host!: HTMLAtomicAriaLiveElement;
-  @State() private message = '';
-  @State() private assertive = false;
+  @State() private regions: Readonly<Regions> = {};
```

Comment:
A re-render is only triggered when we replace (`=`) the object, so I made it `Readonly` to ensure we don't just mutate it.